### ISSUE TITLE
💅 `noUnresolvedImports` ignores ambient `declare module` declarations, and reports the declarations themselves

### ISSUE THREAD

**Issue by aykhanhuseyn** [biomejs/biome#11745]
### Environment information

<details>

```bash
CLI:
  Version:                      2.5.13
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_DISTRIBUTION:           npm
  BIOME_LOG_FILE:               unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v22.18.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         bun/1.4.0

Biome Configuration:
  Status:                       Loaded successfully.
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false
  HTML full support enabled:    unset

Linter:
  JavaScript enabled:           unset
  JSON enabled:                 unset
  CSS enabled:                  unset
  GraphQL enabled:              unset
  Recommended:                  unset
  Enabled rules:
    (full list elided for length: biome.json sets linter.domains project/react/test/types
    to "all", plus correctness/noUndeclaredDependencies and correctness/noUnresolvedImports
    as "error")

Workspace:
  Open Documents:               0
```

</details>

### Rule name

correctness/noUnresolvedImports (and correctness/noUndeclaredDependencies)

### Playground link

https://github.com/aykhanhuseyn/biome-ts-report (not reproducible in the playground: needs a .d.ts, a package.json and multiple files)

### What happens

A module that exists only at runtime is typed with an ambient declaration — the standard TypeScript escape hatch. In the repro these are Module Federation remotes (`@module-federation/vite` has no working remote type generation, so the declarations are hand-written), but the shape is the same for any virtual module.

`host/src/types/remote-app.d.ts`:

```ts
declare module "remote_app/Button" {
  export type ButtonProps = { label: string; onClick?: () => void }
  const Button: (props: ButtonProps) => import("react").ReactElement
  export default Button
}

declare module "remote_app/greet" {
  export function greet(name: string): string
}
```

**1. The declarations are not consulted when resolving imports.**

```
host/src/App.tsx:2:26 lint/correctness/noUnresolvedImports ━━━━━━━━━━━━━━━━━━━━━━━━━━

  × The import specifier remote_app/Button cannot be resolved: module not found

  > 2 │ import RemoteButton from 'remote_app/Button'
      │                          ^^^^^^^^^^^^^^^^^^^

host/src/App.tsx:2:26 lint/correctness/noUndeclaredDependencies ━━━━━━━━━━━━━━━━━━━━━

  × Dependency remote_app isn't specified in host/package.json.
```

**2. The declaration sites are themselves linted as imports.**

```
host/src/types/remote-app.d.ts:1:16 lint/correctness/noUnresolvedImports ━━━━━━━━━━━━

  × The import specifier remote_app/Button cannot be resolved: module not found

  > 1 │ declare module "remote_app/Button" {
      │                ^^^^^^^^^^^^^^^^^^^
```

`tsc -b` passes on the same tree.

#2123 fixed this for the wildcard form (`declare module "*.svg"`); the literal form still reports.

### Expected result

- A `declare module "x"` specifier is a declaration, not an import. `noUnresolvedImports` should never try to resolve it (problem 2 — this one looks like a straightforward bug).
- A literal `declare module "x"` visible to the project should make `x` resolvable for `noUnresolvedImports` and exempt from `noUndeclaredDependencies` (problem 1) — that is what TypeScript itself does, and declaring the module is the only way to express a runtime-only module at all.

This isn't Module Federation-specific. Anything virtual lands here: Vite virtual modules, generated remotes, build-time injected modules. Today the two rules can't be enabled on such a project, because the file written specifically to explain the module to the type system is itself reported as broken.

If treating ambient declarations as a resolution source is too large a change, an option on these rules for ignored specifier patterns would cover the practical case.

### Reproduction

<https://github.com/aykhanhuseyn/biome-ts-report>

```bash
bun install && bun install --cwd remote && bun install --cwd host
bun run check
```

The same repo also surfaces #11744 and #7644; they are independent of this one.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
I appreciate the reproduction, but the LLM put too much text that I fail to understand what are the expected results.

Be more concise and precise, without all the jargon around it
